core/src/wallet/ripple/explorers/NodeRippleLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,282 @@
+                    });
+        }
+
+        Future<String> NodeRippleLikeBlockchainExplorer::pushTransaction(const std::vector<uint8_t> &transaction) {
+            return pushLedgerApiTransaction(transaction);
+        }
+
+        FuturePtr<RippleLikeBlockchainExplorer::TransactionsBulk>
+        NodeRippleLikeBlockchainExplorer::getTransactions(const std::vector<std::string> &addresses,
+                                                          Option<std::string> fromBlockHash,
+                                                          Option<void *> session) {
+            if (addresses.size() != 1) {
+                throw make_exception(api::ErrorCode::INVALID_ARGUMENT,
+                                     "Can only get transactions for 1 address from Ripple Node, but got {} addresses", addresses.size());
+            }
+            NodeRippleLikeBodyRequest bodyRequest;
+            bodyRequest.setMethod("account_tx");
+            bodyRequest.pushParameter("account", addresses[0]);
+            auto requestBody = bodyRequest.getString();
+            return _http->POST("", std::vector<uint8_t>(requestBody.begin(), requestBody.end()))
+                    .template json<TransactionsBulk, Exception>(
+                            LedgerApiParser<TransactionsBulk, RippleLikeTransactionsBulkParser>())
+                    .
+                            template mapPtr<TransactionsBulk>(getExplorerContext(), [fromBlockHash](
+                            const Either<Exception, std::shared_ptr<TransactionsBulk>> &result) {
+                        if (result.isLeft()) {
+                            if (fromBlockHash.isEmpty()) {
+                                throw result.getLeft();
+                            } else {
+                                throw make_exception(api::ErrorCode::BLOCK_NOT_FOUND,
+                                                     "Unable to find block with hash {}", fromBlockHash.getValue());
+                            }
+                        } else {
+                            return result.getRight();
+                        }
+                    });
+        }
+
+        FuturePtr<Block> NodeRippleLikeBlockchainExplorer::getCurrentBlock() const {
+            NodeRippleLikeBodyRequest bodyRequest;
+            bodyRequest.setMethod("ledger");
+            bodyRequest.pushParameter("ledger_index", std::string("validated"));
+            auto requestBody = bodyRequest.getString();
+            return _http->POST("", std::vector<uint8_t>(requestBody.begin(), requestBody.end()))
+                    .template json<Block, Exception>(LedgerApiParser<Block, RippleLikeBlockParser>())
+                    .
+                            template mapPtr<Block>(getExplorerContext(),
+                                                   [](const Either<Exception, std::shared_ptr<Block>> &result) {
```

Comment:
I like that pattern, and that’s just `Result::unwrap()` in Rust (on a left, it panics, which is somewhat the same as an exception, but we don’t like exceptions in Rust). Maybe we can add that to `Either` and use it here? :)

Reply:
Great idea !